### PR TITLE
Add support for running tests as native executables.

### PR DESCRIPTION
**Pull request by ncbray**

### CODE REVIEW

File: scripts/run.py [hunk 1]
```diff
@@ -23,14 +24,25 @@ class Error(Exception):
...
+  if process.returncode != 0:
+    raise Error(stderr)
+
+
 def main(args):
   parser = argparse.ArgumentParser()
   parser.add_argument('c_file', help='test C file')
-  parser.add_argument('js_file', nargs='?',
-                      help='test JS file', default=WASM_JS)
+  parser.add_argument('--js-env', help='JS/WASM environment file file')
```
Comment (dschuff) [inline scripts/run.py]: file file?

File: scripts/run.py [hunk 2]
```diff
@@ -40,19 +52,24 @@ def main(args):
...
+      subprocess.check_call([WACC, '-fno-builtin', options.c_file, '-o', s_file, '-I', '.'])
+      subprocess.check_call([sys.executable, WASMATE, s_file, '-o', wast_file])
+      subprocess.check_call([SEXPR_WASM, wast_file, '-o', wasm_file])
+      call_and_suppress_stderr([D8, options.js_env, '--', wasm_file])
+
+    if options.native_env:
+      executable_file = os.path.join(temp_dir, basename)
+      subprocess.check_call([CLANG, '-O2', options.c_file, options.native_env,
+                             '-o', executable_file, '-I', '.'])
+      call_and_suppress_stderr([executable_file])
```
Comment (dschuff) [inline scripts/run.py]: maybe we'd want to keep stderr for the native file, since we don't expect stderr spam.

File: scripts/run.py (hunk 1 above)
Comment (binji) [inline scripts/run.py]: "environment" makes me think of `os.environ` -- something else?

**Comment by dschuff** [comment]
Otherwise LGTM

File: scripts/run.py (hunk 1 above)
Comment (ncbray) [inline scripts/run.py]: I am open to suggestions, but I am also out of ideas...

Comment (binji) [inline scripts/run.py]: Me either. So LGTM

File: scripts/run.py [hunk 3]
```diff
@@ -23,14 +24,25 @@ class Error(Exception):
   pass
 
 
+def call_and_suppress_stderr(cmd):
+  process = subprocess.Popen(cmd,
+                              stderr=subprocess.PIPE)
```
Comment (jfbastien) [inline scripts/run.py]: Extra whitespace?

**Pull request merged**